app/services/control_rate_service.rb

```diff
@@ -61,10 +61,9 @@ def fetch_all_data
 
     results.controlled_patients_rate = repository.controlled_rates[slug]
     results.uncontrolled_patients_rate = repository.uncontrolled_rates[slug]
-
-    results.calculate_percentages(:controlled_patients, with_ltfu: true)
-    results.calculate_percentages(:uncontrolled_patients, with_ltfu: true)
-    results.calculate_percentages(:ltfu_patients)
+    results.controlled_patients_with_ltfu_rate = repository.controlled_rates(with_ltfu: true)[slug]
+    results.uncontrolled_patients_with_ltfu_rate = repository.uncontrolled_rates(with_ltfu: true)[slug]
+    results.ltfu_patients_rate = repository.ltfu_rates[slug]
```

Comment:
This is just switching to the more explicit way of setting these repo keys? Consistent with above?

Reply:
Yup, and its removing the responsibility from the Result object for calculating these rates.